src/Microsoft.ML.StandardLearners/Standard/Online/OnlineLinear.cs

```diff
@@ -253,7 +253,11 @@ private protected sealed override TModel TrainModelCore(TrainContext context)
         {
             Host.CheckValue(context, nameof(context));
             var initPredictor = context.InitialPredictor;
-            var initLinearPred = initPredictor as LinearModelParameters ?? (initPredictor as CalibratedPredictorBase)?.SubPredictor as LinearModelParameters;
+            LinearModelParameters initLinearPred = null;
+            if (initPredictor is LinearModelParameters)
+                initLinearPred = (LinearModelParameters)initPredictor;
+            else if (initPredictor is CalibratedPredictorBase<LinearBinaryModelParameters, PlattCalibrator>)
+                initLinearPred = ((CalibratedPredictorBase<LinearBinaryModelParameters, PlattCalibrator>)initPredictor).SubPredictor;
             Host.CheckParam(initPredictor == null || initLinearPred != null, nameof(context), "Not a linear predictor.");
```

Comment:
>Host.CheckParam(initPredictor == null || initLinearPred != null, nameof(context), "Not a linear predictor."); [](start = 12, length = 109)

This is effectively repeating the test already in the prior if statements. So this can be improved by just using `else throw Host.ExceptParam(...)` and so on.